cmd/sourcemaps/upload.go

```diff
@@ -0,0 +1,398 @@
+		RunE:  runE(client),
+	}
+
+	cmd.SetUsageTemplate(resourcescmd.SubcommandUsageTemplate())
+	initFlags(cmd)
+
+	return cmd
+}
+
+func runE(client resources.Client) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		u, _ := url.JoinPath(
+			viper.GetString(cliflags.BaseURIFlag),
+			"api/v2/caller-identity",
+		)
+		res, err := client.MakeRequest(
+			viper.GetString(cliflags.AccessTokenFlag),
+			"GET",
+			u,
+			"application/json",
+			nil,
+			nil,
+			false,
+		)
+		if err != nil {
+			return output.NewCmdOutputError(err, viper.GetString(cliflags.OutputFlag))
+		}
+
+		var result struct{ AccountID string }
+		if err = json.Unmarshal(res, &result); err != nil {
+			return output.NewCmdOutputError(err, viper.GetString(cliflags.OutputFlag))
+		}
+
+		projectKey := viper.GetString(cliflags.ProjectFlag)
+		u, _ = url.JoinPath(
+			viper.GetString(cliflags.BaseURIFlag),
+			"api/v2/projects",
+			projectKey,
+		)
+		res, err = client.MakeRequest(
+			viper.GetString(cliflags.AccessTokenFlag),
+			"GET",
+			u,
+			"application/json",
+			nil,
+			nil,
+			false,
+		)
+		if err != nil {
+			return output.NewCmdOutputError(err, viper.GetString(cliflags.OutputFlag))
+		}
+
+		var projectResult struct {
+			Items []struct {
+				ID string `json:"_id"`
+			}
+		}
+		if err = json.Unmarshal(res, &projectResult); err != nil {
+			return output.NewCmdOutputError(err, viper.GetString(cliflags.OutputFlag))
+		}
+
+		appVersion := viper.GetString(appVersionFlag)
+		path := viper.GetString(pathFlag)
+		basePath := viper.GetString(basePathFlag)
+		backendUrl := viper.GetString(backendUrlFlag)
+
+		if backendUrl == "" {
+			backendUrl = defaultBackendUrl
+		}
+
+		highlightKey, projectID, err := verifyApiKey(result.AccountID, projectResult.Items[0].ID, backendUrl)
+		if err != nil {
+			return fmt.Errorf("failed to verify API key: %v", err)
+		}
+
+		fmt.Printf("Starting to upload source maps from %s\n", path)
+
+		files, err := getAllSourceMapFiles(path)
+		if err != nil {
+			return fmt.Errorf("failed to find sourcemap files: %v", err)
+		}
+
+		if len(files) == 0 {
+			return fmt.Errorf("no source maps found in %s, is this the correct path?", path)
+		}
+
+		s3Keys := make([]string, 0, len(files))
+		for _, file := range files {
+			s3Keys = append(s3Keys, getS3Key(projectID, appVersion, basePath, file.Name))
+		}
+
+		uploadUrls, err := getSourceMapUploadUrls(highlightKey, s3Keys, backendUrl)
+		if err != nil {
+			return fmt.Errorf("failed to get upload URLs: %v", err)
+		}
+
+		for i, file := range files {
+			if err := uploadFile(file.Path, uploadUrls[i], file.Name); err != nil {
+				return fmt.Errorf("failed to upload file %s: %v", file.Path, err)
+			}
+		}
+
+		fmt.Println("Successfully uploaded all sourcemaps")
+		return nil
+	}
+}
+
+func verifyApiKey(accountID, projectID, backendUrl string) (string, string, error) {
```

Comment:
It is unclear what the returned strings represent. Add documentation to the this method to clarify it.